crates/subspace-networking/src/constructor.rs

```diff
@@ -482,8 +484,12 @@ where
 
     debug!(?connection_limits, "DSN connection limits set.");
 
-    let enable_autonat = external_addresses.is_empty() && kademlia_mode.is_dynamic();
-    debug!(%enable_autonat, ?external_addresses, ?kademlia_mode, "Autonat settings.");
+    let autonat_boot_delay = match kademlia_mode {
+        KademliaMode::Static(_) => AUTONAT_SERVER_PROBE_DELAY,
+        KademliaMode::Dynamic => AutonatConfig::default().boot_delay,
+    };
```

Comment:
This is not equivalent to before. If `KademliaMode::Dynamic`, but we have external addresses, we don't need autonat either, that is what the `enable_autonat` checked before. We just need `if`/`else` after that.